DataFormats/Headers/include/Headers/DataHeader.h

```diff
@@ -25,6 +25,7 @@
 #include <iostream>
 #include <memory>
 #include <cassert>
+#include <cstring>
```

Comment:
Indeed, memcmp decays to pretty efficient code, (sometimes) marginally less effective than direct comparisons.